src/secondary_tokio_runtime.rs

```diff
@@ -0,0 +1,117 @@
+use std::{future::Future, pin::Pin};
+
+use tokio::{
+    runtime::Handle,
+    select,
+    sync::mpsc::{error::TrySendError, Sender},
+};
+
+use crate::{error::Error, ComputeHeavyFutureExecutor};
+
+const DEFAULT_SECONDARY_EXECUTOR_NICENESS: i8 = 10;
+
+fn default_secondary_executor_thread_count() -> usize {
+    num_cpus::get()
+}
+
+type BackgroundFuture = Pin<Box<dyn Future<Output = ()> + Send + 'static>>;
+
+pub(crate) struct SecondaryTokioRuntimeExecutor {
+    tx: Sender<BackgroundFuture>,
+}
+
+impl SecondaryTokioRuntimeExecutor {
+    pub(crate) fn new(niceness: Option<i8>, thread_count: Option<usize>) -> Self {
```

Comment:
Note: also add a customizable queue size for the background job queue, also concurrency limit on running tasks.

Also look at adding cancellable behavior.